optuna/multi_objective/hypervolume/base.py

```diff
@@ -0,0 +1,47 @@
+import abc
+
+import numpy as np
+
+
+class BaseHypervolume(object, metaclass=abc.ABCMeta):
```

Comment:
Do you think it's better to make the classes/the hypervolume directory public? If it's only intended for MOTPE for the time being, it could be private and I also think we can reduce the number of "validations".

Reply:
I agree with the idea of making the hypervolume class private. This would reduce the validation function and the overall amount of code.